sycl/test/basic_tests/image_accessor_readwrite_half.cpp

```diff
@@ -0,0 +1,162 @@
+    });
+  }
+}
+
+template <typename ReadDataT, s::image_channel_type ImgType>
+void check_read_type_order(char *HostPtr, const s::image_channel_order ImgOrder,
+                           ReadDataT ExpectedColor) {
+
+  int Coord(2);
+  ReadDataT ReadData;
+  {
+    // image with dim = 1
+    s::image<1> Img(HostPtr, ImgOrder, ImgType, s::range<1>{10});
+    s::queue Queue;
+    s::buffer<ReadDataT, 1> ReadDataBuf(&ReadData, s::range<1>(1));
+    Queue.submit([&](s::handler &cgh) {
+      auto ReadAcc = Img.get_access<ReadDataT, s::access::mode::read>(cgh);
+      s::accessor<ReadDataT, 1, s::access::mode::write> ReadDataBufAcc(
+          ReadDataBuf, cgh);
+
+    cgh.single_task<class kernel_class<ReadDataT, static_cast<int>(ImgType), 1>>([=](){
+      ReadDataT RetColor = ReadAcc.read(Coord);
+      ReadDataBufAcc[0] = RetColor;
+    });
+    });
+  }
+  check_read_data(ReadData, ExpectedColor);
+}
+
+void check_half4(char *HostPtr) {
+
+  // Calling only valid channel types with s::cl_half4.
+  // s::image_channel_type::snorm_int8,
+  write_type_order<s::cl_half4, s::image_channel_type::snorm_int8>(
+      HostPtr, s::image_channel_order::rgba, s::cl_half4(2, -2, 0.375f, 0));
+  check_read_type_order<s::cl_half4, s::image_channel_type::snorm_int8>(
+      HostPtr, s::image_channel_order::rgba,
+      s::cl_half4(1, -1, ((float)48 / 127) /*0.3779527544975280762f*/, 0));
+
+  // s::image_channel_type::snorm_int16,
+  write_type_order<s::cl_half4, s::image_channel_type::snorm_int16>(
+      HostPtr, s::image_channel_order::rgba, s::cl_half4(2, -2, 0.375f, 0));
+  check_read_type_order<s::cl_half4, s::image_channel_type::snorm_int16>(
+      HostPtr, s::image_channel_order::rgba,
+      s::cl_half4(1, -1, ((float)12288 / 32767) /*0.375011444091796875f*/, 0));
+
+  // s::image_channel_type::unorm_int8,
+  write_type_order<s::cl_half4, s::image_channel_type::unorm_int8>(
+      HostPtr, s::image_channel_order::rgba, s::cl_half4(2, -2, 0.375f, 0));
+  check_read_type_order<s::cl_half4, s::image_channel_type::unorm_int8>(
+      HostPtr, s::image_channel_order::rgba,
+      s::cl_half4(1, 0, ((float)96 / 255) /*0.3764705955982208252f*/, 0));
+
+  // s::image_channel_type::unorm_int16
+  write_type_order<s::cl_half4, s::image_channel_type::unorm_int16>(
+      HostPtr, s::image_channel_order::rgba, s::cl_half4(1, -1, 0.375f, 0));
+  check_read_type_order<s::cl_half4, s::image_channel_type::unorm_int16>(
+      HostPtr, s::image_channel_order::rgba,
+      s::cl_half4(1, 0, ((float)24576 / 65535) /*0.3750057220458984375f*/, 0));
+
+  // s::image_channel_type::fp16
+  write_type_order<s::cl_half4, s::image_channel_type::fp16>(
+      HostPtr, s::image_channel_order::rgba, s::cl_half4(2, -2, 0.375f, 0));
+  check_read_type_order<s::cl_half4, s::image_channel_type::fp16>(
+      HostPtr, s::image_channel_order::rgba, s::cl_half4(2, -2, 0.375f, 0));
+};
+
+int main() {
+  // Checking only for dimension=1.
+  // create image:
+  char HostPtr[100];
```

Comment:
Suggest using std::array with size = sizeof(sycl::cl_half4). And avoid using magic number on line 99 and 80.